networkx/algorithms/tests/test_hierarchy_coordinates.py

```diff
@@ -0,0 +1,225 @@
+    compare_hashed_graphs(
+        pruned_from_bottom[1], nx.DiGraph(layer_2 + layer_1 + layer_0)
+    )
+    compare_hashed_graphs(pruned_from_bottom[2], nx.DiGraph(layer_1 + layer_0))
+    compare_hashed_graphs(pruned_from_bottom[3], nx.DiGraph(layer_0))
+
+    G = nx.cycle_graph(5, create_using=nx.DiGraph)  # No cycles
+    pytest.raises(nx.NetworkXError, nx.recursive_leaf_removal, G)
+
+
+def test_weight_nodes_by_condensation():
+    center_cycle = [(0, 1), (1, 2), (2, 3), (3, 1), (3, 4)]
+    G = nx.DiGraph(center_cycle)  # unit weights by default
+    pytest.raises(
+        nx.NetworkXError, nx.weight_nodes_by_condensation, G
+    )  # Must be condensed (acyclic)
+
+    condensed_G = nx.condensation(G)
+    nodes_w_cycle_weight = nx.weight_nodes_by_condensation(condensed_G)
+    assert nodes_w_cycle_weight.nodes[0]["weight"] == 1
+    assert nodes_w_cycle_weight.nodes[1]["weight"] == 3
+    assert nodes_w_cycle_weight.nodes[2]["weight"] == 1
+
+
+def test_node_weighted_condense():
+    center_cycle_A = np.array(
+        [
+            [0, 1, 0, 0, 0, 0, 0],
+            [0, 0, 1, 0, 0, 0, 0],
+            [0, 0, 0, 1, 0, 0, 0],
+            [0, 1, 0, 0, 1, 0, 0],
+            [0, 0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0, 1],
+            [0, 0, 0, 0, 0, 1, 0],
+        ]
+    )  # this condensation eliminates the smaller, unconnected cycles, e.g. {(5, 6), (6, 5)}
+    center_cycle = [(0, 1), (1, 2), (2, 3), (3, 1), (3, 4), (5, 6), (6, 5)]
+    G = nx.DiGraph(center_cycle)  # unit weights by default
+    compare_hashed_graphs(
+        nx.from_numpy_array(A=center_cycle_A, create_using=nx.DiGraph), G
+    )
+
+    condensed_G, original_G = nx.node_weighted_condense(A=center_cycle_A)
+    assert (
+        len(condensed_G) == 1 == len(original_G)
+    )  # single threshold for binary graphs
+    assert condensed_G[0].nodes[0]["weight"] == 1
+    assert condensed_G[0].nodes[1]["weight"] == 3
+    assert condensed_G[0].nodes[2]["weight"] == 1
+
+    # Weighted network condense:
+    weighted_center_cycle_A = np.array(
+        [
+            [0, 0.9, 0, 0, 0, 0, 0],
+            [0, 0, 0.8, 0, 0, 0, 0],
+            [0, 0, 0, 0.4, 0, 0, 0],
+            [0, 1.7, 0, 0, 0.6, 0, 0],
+            [0, 0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0, 0.4],
+            [0, 0, 0, 0, 0, 0.2, 0],
+        ]
+    )  # this condensation eliminates the smaller, unconnected cycles, e.g. {(5, 6), (6, 5)}
+
+    condensed_Gs, binary_Gs = nx.node_weighted_condense(
+        A=weighted_center_cycle_A, num_thresholds=5, threshold_distribution=None
+    )
+    assert len(condensed_Gs) == len(
+        binary_Gs
+    )  # Not necessarily equal to num thresholds, as empty graphs are dropped
+    for index in range(len(binary_Gs)):
+        assert np.array_equal(
+            np.unique(nx.to_numpy_array(binary_Gs[index])), [0, 1]
+        )  # binary checks
+        assert np.array_equal(np.unique(nx.to_numpy_array(condensed_Gs[index])), [0, 1])
```

Comment:
```suggestion
    for binaryG, condensedG in zip(binary_Gs, condensed_Gs):
        # binary checks
        assert np.array_equal(np.unique(nx.to_numpy_array(binaryG)), [0, 1])
        assert np.array_equal(np.unique(nx.to_numpy_array(condensedG)), [0, 1])
```
You could easily split this into two loops also.   Other places with `index` can use `zip` as well.
More readable, should be faster because done in C, more "pythonic" (whatever that means).

Reply:
I implemented the suggested changes, though I didn't find other places which could better use `zip`; I think you might've caught them all already.